packages/@aws-cdk/aws-iam/README.md

```diff
@@ -238,6 +238,36 @@ const newPolicyDocument = PolicyDocument.fromJson(policyDocument);
 
 ```
 
+### OpenID Connect Providers
+
+OIDC identity providers are entities in IAM that describe an external identity
+provider (IdP) service that supports the [OpenID Connect] (OIDC) standard, such as
+Google or Salesforce. You use an IAM OIDC identity provider when you want to
+establish trust between an OIDC-compatible IdP and your AWS account. This is
+useful when creating a mobile app or web application that requires access to AWS
+resources, but you don't want to create custom sign-in code or manage your own
+user identities. For more information about this scenario, see [About Web
+Identity Federation].
+
+[OpenID Connect]: http://openid.net/connect
+[About Web Identity Federation]: https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_oidc.html
+
+The following examples defines an OpenID Connect provider. Two client IDs
+(audiences) are will be able to send authentication requests to
+https://openid/connect.
+
+```ts
+new OpenIdConnectProvider(this, 'MyProvider', {
```

Comment:
Any guidance on how we use this provider?

I know you're just building it for cdk8s but you seem to be implying we can tie this to AWS identities in some way...?

Reply:
I added an example on how to use with cognito's `CfnIdentityPool`. Once we have an L2, we will need to switch that up (copy @nija-at).